### PR TITLE
Make the package actually usable

### PR DESCRIPTION
**Pull request by JamesWrigley**
TL;DR everything's renamed and all the tests pass, plus the `SSHManager` tests have been enabled by default on 64bit Linux/OSX since we can use LibSSH.jl to create the SSH server it needs. The SSH tests are kinda heavy though, I don't really know why they create 9 workers by default. If that's annoying I'd be open to lowering it to like 6 or so in the future.

The only important change is that the `project` argument must be set correctly if using the `SSHManager`. I've set it to the currently active project by default, which should at least be a sensible choice for clusters with shared filesystems. I also added a changelog so we can track changes from upstream.

I've tried to keep all the commits atomic, so I'd recommend reviewing the PR commit-by-commit.

### CODE REVIEW

**Comment by JamesWrigley** [comment]
> TL;DR everything's renamed and all the tests pass

Lol, I spoke too soon.

**Review by jpsamaroo — APPROVED** [review]
Awesome work! :champagne:
